Fundamentals/Design/Stylesheets-Javascript/index-v9.md

```diff
@@ -0,0 +1,148 @@
+
+![Linking CSS in template](images/4-link-css-v9.png)
+
+By default, the stylesheets will be saved in the `wwwroot/css` folder in the solution.
+To reference them you can use either of the methods used in the above screenshot.
+
+```html
+<link rel="stylesheet" href='@Url.Content("~/css/mystylesheet.css")' />
+```
+or
+```html
+<link rel="stylesheet" href="/css/mystylesheet.css" />
+```
+
+With the stylesheet referenced, you will be able to style the template file with the rules and classes defined in the stylesheet.
+
+Your stylesheets can be used in Rich Text Editors (datatype) as well - please see the [Rich Text Editor](../../Backoffice/Property-Editors/Built-in-Property-Editors/Rich-Text-Editor/RTE-Styles/index.md) documentation for more information.
+
+:::note
+If your RTE is styled differently on the frontend of the site, the backoffice styling might be getting overwritten by other stylesheets you have included.
+:::
+
+## JavaScript files in the Backoffice
+
+To create and edit JavaScript files in the Backoffice, head on over to the Scripts folder in the Settings section of the Backoffice.
+
+![Creating a new JavaScript](images/8-create-js.png)
+
+From here you can add a new JavaScript file, or a new folder.
+
+Add a new JavaScript file and write your code:
+
+![Sample JS script](images/9-myscript.png)
+
+Then, navigate to the template where you would like to include your JS file.
+
+```html
+<script src="/scripts/myScript.js"></script>
+```
+
+![Reference the script in template](images/10-reference-script-v9.png)
+
+By default all JavaScript files will be stored in the `wwwroot/scripts` folder in the solution.
+
+:::tip
+If you are working locally, you can create CSS and JS files outside of the Backoffice - long as they are placed in appropiate folders (`css` and `scripts`), they will show up in the Backoffice.
+:::
+
+## Bundling & Minification for JavaScript and CSS
+
+You can use whichever tool you are comfortable with for bundling & minification by implementing the `IRuntimeMinifier` interface in your custom minifier class, though it is worth noting that Umbraco 9 ships with Smidge which offers lightweight runtime bundling and minification.
+
+You can create various bundles of your site's CSS or JavaScript files in your code that can be rendered later in your views. There can be a single bundle for the entire site, or a common bundle for the files you want to be loaded on every page, as well as page-specific bundles, just by listing your resources in the order you like.
+```csharp
+_runtimeMinifier.CreateJsBundle("inline-js-bundle",
+                    BundlingOptions.NotOptimizedAndComposite,
+                    new[] { "~/scripts/myScript1.js", "~/scripts/myScript2.js" });
+
+_runtimeMinifier.CreateCssBundle("inline-css-bundle",
+                    BundlingOptions.NotOptimizedAndComposite,
+                    new[] { "~/css/mystylesheet.css" });
+
+```
```

Comment:
I think it is a good idea to be clear where this code lives, just to make sure that there is no confusion 🙂

So the question is, where does this code live?

Reply:
Yeah exactly. 

I'll be honest I'm not sure what would be the best place to do it, but taking a look at CMS codebase we do it lazily on the first backoffice request in some custom middleware, however, I think that might be a bit overkill for the docs. But we cannot do it in the composer either, since DI has not been set up there yet. My best guess would be that it should be done in a component, then you can inject the `IRunTimeMinifier` and do it in the `Initialize` method. It's been a while since I looked at this, but I think that's how I did when testing.

Reply:
Just did a quick test to make sure that it would work with a component and it does, I used this code which seems to work just fine

```C#
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.WebAssets;

namespace Umbraco.Cms.Web.UI
{
    public class MyComponent : IComponent
    {
        private readonly IRuntimeMinifier _runtimeMinifier;

        public MyComponent(IRuntimeMinifier runtimeMinifier)
        {
            _runtimeMinifier = runtimeMinifier;
        }

        public void Initialize()
        {
            _runtimeMinifier.CreateJsBundle("inline-js-bundle",
                BundlingOptions.NotOptimizedAndComposite,
                new[] { "~/scripts/script1.js", "~/scripts/script2.js" });

            _runtimeMinifier.CreateCssBundle("inline-css-bundle",
                BundlingOptions.NotOptimizedAndComposite,
                new[] { "~/css/style1.css", "~/css/style2.css" });
        }

        public void Terminate(){}
    }

    public class MyComposer : ComponentComposer<MyComponent>
    {

    }
}
```

Reply:
Yes, indeed how I also was testing it, but it is been a while, so thanks for the code example @nikolajlauridsen 🙌